.github/workflows/bench.yml

```diff
@@ -0,0 +1,73 @@
+name: BENCHMARK
+
+# Controls when the action will run.
+on:
+  pull_request:
+    branches: testnet
+
+jobs:
+  hardware-benchmarks:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          echo "HOME=/home/actions" >> $GITHUB_ENV
+
+      - name: Inspect benchmark CI config
+        run: |
+          PALLETS_IN_CI_CONFIG=$(grep "PALLET:" .github/workflows/bench.yml | tr -d ' ' | cut -f2 -d: | sort | xargs)
+          PALLETS_IN_SOURCE=$(ls -1 pallets/ | sort | xargs)
+          if [ "$PALLETS_IN_CI_CONFIG" != "$PALLETS_IN_SOURCE" ]; then
+              echo "FAIL: Pallet mismatch between source code and benchmark CI config"
+              exit 1
+          fi
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2022-03-21
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+        continue-on-error: true
+
+      - name: Difficulty
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: difficulty
+        run: |
+          ./scripts/bench.sh -p $PALLET -cb
+
+      - name: Rewards
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: rewards
+        run: |
+          ./scripts/bench.sh -p $PALLET -cb
+
+      # Run creditcoin pallet with lesser steps
+      - name: Creditcoin
+        shell: bash
+        continue-on-error: true
+        env:
+          STEPS: 8
+          REPEAT: 8
```

Comment:
From https://github.com/gluwa/creditcoin/pull/177#discussion_r869284890

```
repeat is at least 30.
steps is at least 4.
```

Why go with 8 ?

Reply:
In my last discussion with @CAGS295 , we settled for 8 steps and repeats to reduce time for `on_initialize` .